core/src/tile/tileWorker.cpp

```diff
@@ -64,43 +63,51 @@ void TileWorker::run() {
             // Pop highest priority tile from queue
             auto it = std::min_element(m_queue.begin(), m_queue.end(),
                 [](const auto& a, const auto& b) {
-                    if (a->tile->isVisible() != b->tile->isVisible()) {
-                        return a->tile->isVisible();
+                    if (a->isProxy() != b->isProxy()) {
+                        return !a->isProxy();
                     }
-                    if (a->tile->sourceID() == b->tile->sourceID() &&
-                        a->tile->sourceGeneration() != b->tile->sourceGeneration()) {
-                        return a->tile->sourceGeneration() < b->tile->sourceGeneration();
+                    if (a->source().id() == b->source().id() &&
+                        a->sourceGeneration() != b->sourceGeneration()) {
+                        return a->sourceGeneration() < b->sourceGeneration();
                     }
-                    return a->tile->getPriority() < b->tile->getPriority();
+                    return a->getPriority() < b->getPriority();
                 });
 
             task = std::move(*it);
             m_queue.erase(it);
         }
 
-        if (task->tile->isCanceled()) {
-            continue;
-        }
-
-        auto tileData = task->process();
-
-        // NB: Save shared reference to Scene while building tile
-        auto scene = m_tileManager.getScene();
+        if (task->isCanceled()) { continue; }
 
+        // Save shared reference to Scene while building tile
+        // FIXME: Scene could be released on Worker-Thread and
+        // therefore call unsafe glDelete* functions...
+        auto scene = m_scene;
```

Comment:
Unfortunately this would only move the problem: To use the scene it is necessary to acquire a shared reference.